README.md

```diff
@@ -2,31 +2,71 @@
-[![GitHub Workflow Release Status](https://img.shields.io/github/workflow/status/edaa-org/pyEDAA.ProjectModel/Release?label=release&logo=GitHub%20Actions&logoColor=FFFFFF)](https://github.com/edaa-org/pyEDAA.ProjectModel/actions?query=workflow%3A%22Release%22)
 [![PyPI](https://img.shields.io/pypi/v/pyEDAA.ProjectModel?logo=PyPI&logoColor=FBE072)](https://pypi.org/project/pyEDAA.ProjectModel/)
 ![PyPI - Status](https://img.shields.io/pypi/status/pyEDAA.ProjectModel?logo=PyPI&logoColor=FBE072)
 ![PyPI - Python Version](https://img.shields.io/pypi/pyversions/pyEDAA.ProjectModel?logo=PyPI&logoColor=FBE072)
 [![Libraries.io status for latest release](https://img.shields.io/librariesio/release/pypi/pyEDAA.ProjectModel)](https://libraries.io/github/edaa-org/pyEDAA.ProjectModel)
 [![Requires.io](https://img.shields.io/requires/github/edaa-org/pyEDAA.ProjectModel)](https://requires.io/github/edaa-org/pyEDAA.ProjectModel/requirements/?branch=main)  
-[![GitHub Workflow - Documentation Status](https://img.shields.io/github/workflow/status/edaa-org/pyEDAA.ProjectModel/Documentation?label=documentation&logo=GitHub%20Actions&logoColor=FFFFFF)](https://github.com/edaa-org/pyEDAA.ProjectModel/actions?query=workflow%3A%22Documentation%22)
 [![Documentation License](https://img.shields.io/badge/doc%20license-CC--BY%204.0-green)](LICENSE.md)
 [![Documentation - Read Now!](https://img.shields.io/badge/doc-read%20now%20%E2%9E%94-blueviolet)](https://edaa-org.github.io/pyEDAA.ProjectModel/)
 
+<!--
+[![Dependent repos (via libraries.io)](https://img.shields.io/librariesio/dependent-repos/pypi/pyEDAA.ProjectModel?logo=GitHub)](https://github.com/edaa-org/pyEDAA.ProjectModel/network/dependents)
+-->
+
 # pyEDAA.ProjectModel
 
-* abstract model of EDA tool projects
-* filesets, filetypes, ...
+This package provides a unified abstract project model for HDL designs and EDA tools.
+Third-party frameworks can derive own classes and implement additional logic to create
+a concrete project model for their tools.
+
+Frameworks consuming this model can build higher level features and services on top of
+such a model, while supporting multiple input sources.
+
+## Data Model
+
+1. The toplevel element is a `Project`, which contains one or multiple designs.
+2. A `Design` is a variant of a project and contains filesets.
+3. A `FileSet` contains files or further sub-filesets.
+4. A `File` represents a single file. E.g. source files, configuration files, constraint files
+5. A `VHDLLibrary` represents a group of `VHDLSourceFile`s being compiled into the same VHDL library.
+
+![img.png](doc/datamodel.png)
+
+## Features
+
+* Construct a project model:  
+  * top-down (project &rarr; design &rarr; fileset &rarr; file) or
+  * bottom-up (file &rarr; fileset &rarr; design &rarr; project) or
+  * parsing a project file.
```

Comment:
Is it possible to "parsing project file(s)". That is, can we load two XPR files, or combine an XPR file and `.pro` files in the same ProjectModel?

Reply:
A `VivadoProjectFile` and a `OSVVMProjectFile` represent a project file. They offer a `Parse()` method, which fills a property called `ProjectModel`. So currently, each file has it's own project model. I'm not sure how we can define a merge operation an them.

Maybe that would be the case for the parent project:
```py
project = Project("Parent", rootDirectory=Path.cwd())
osvvmFileSet = FileSet("OSVVM", project=project)
osvvmProjectFile = OSVVMProjectFile(osvvmPath, fileSet=osvvmFileSet)
osvvmProjectFile.Parse()

osvvmProject = osvvmProjectFile.ProjectModel
```

Maybe, we can have a
```py
osvvmFileSet.Merge(osvvmProject)
```
or so?

Reply:
I believe it would be desirable to have a Merge operation for combining any two projects. I know that many possible combinations will produce conflicts, so I think we don't need a "too clever" approach but just produce meaningful failures when such conflicts arise. Then, it's up to users to understand/know what they can combine and what they cannot.

Particular use cases for combining two projects are: OSVVM and XPR in the context of tsfpga, or OSVVM and VUnit's API in the context of VUnit and OSVB.

However, that's out of scope of this PR, so let's discuss it in an upcoming issue/PR.